tests/dom/basictests.cpp

```diff
@@ -221,6 +221,64 @@ namespace number_tests {
     std::vector<char> buf(1024);
     simdjson::dom::parser parser;
 
+    bool is_pow_correct{1e-308 == std::pow(10,-308)};
+    int start_point = is_pow_correct ? -1000 : -307;
+    if(!is_pow_correct) {
+      std::cout << "On your system, the pow function is busted. Sorry about that. " << std::endl;
+    }
+    for (int i = start_point; i <= 308; ++i) {// large negative values should be zero.
+      size_t n = snprintf(buf.data(), buf.size(), "1e%d", i);
+      if (n >= buf.size()) { abort(); }
+      double actual;
+      auto error = parser.parse(buf.data(), n).get(actual);
+      if (error) { std::cerr << error << std::endl; return false; }
+      double expected = ((i >= -307) ? testing_power_of_ten[i + 307]: std::pow(10, i));
+	  // In floating-point arithmetic, -0.0 == 0.0, so compare signs by checking the inverse of the numbers as well
+      if(actual!=expected || (actual == 0.0 && 1.0/actual!=1.0/expected)) {
+        std::cerr << "JSON '" << buf.data() << " parsed to ";
+        fprintf( stderr," %18.18g instead of %18.18g\n", actual, expected); // formatting numbers is easier with printf
+        SIMDJSON_SHOW_DEFINE(FLT_EVAL_METHOD);
+        return false;
+      }
+    }
+    printf("Powers of 10 can be parsed.\n");
+    return true;
+  }
+
+  bool negative_powers_of_ten() {
+    std::cout << __func__ << std::endl;
+    std::vector<char> buf(1024);
+    simdjson::dom::parser parser;
+
+    bool is_pow_correct{-1e-308 == -std::pow(10,-308)};
+    int start_point = is_pow_correct ? -1000 : -307;
+    if(!is_pow_correct) {
+      std::cout << "On your system, the pow function is busted. Sorry about that. " << std::endl;
+    }
+    for (int i = start_point; i <= 308; ++i) {// large negative values should be zero.
+      size_t n = snprintf(buf.data(), buf.size(), "-1e%d", i);
+      if (n >= buf.size()) { abort(); }
+      double actual;
+      auto error = parser.parse(buf.data(), n).get(actual);
+      if (error) { std::cerr << error << std::endl; return false; }
+      double expected = -(((i >= -307) ? testing_power_of_ten[i + 307]: std::pow(10, i)));
+	  // In floating-point arithmetic, -0.0 == 0.0, so compare signs by checking the inverse of the numbers as well
```

Comment:
I forgot to remove the tabs from this line in my commit, I meant for this to be spaces like the below `if` 

In another PR, I proposed adding an `.editorconfig` that editors would pick up automatically to avoid this mistake

Reply:
Fixed.